back/src/chat/privmsg/privmsg.service.ts

```diff
@@ -0,0 +1,78 @@
+import { Injectable, BadRequestException } from '@nestjs/common';
+import { PrismaService } from '@prisma/prisma.service';
+import { PaginationDto } from '@type/pagination.dto';
+import { Channel, User, Relationship, RelationKind, ChannelRole, ChannelKind } from '@prisma/client';
+import { ChatService } from '@chat/chat.service';
+
+@Injectable()
+export class PrivmsgService {
+	constructor(
+		private readonly prisma: PrismaService,
+		private readonly chatService: ChatService
+	) {}
+
+	async getAll(pagination: PaginationDto): Promise<Channel[]> {
+		return await this.prisma.channel.findMany(pagination);
+	}
+
+	async join(user: User, privmsgId: number, socketId: string): Promise<Channel> {
+		let lowUserId, highUserId;
+		if (user.id > privmsgId) {
+			lowUserId = privmsgId;
+			highUserId = user.id;
+		} else {
+			lowUserId = user.id;
+			highUserId = privmsgId;
+		}
+		const convName = lowUserId.toString() + '-' + highUserId.toString();
+		const privConv = await this.prisma.channel.findUnique({
+			where: { name: convName },
+			include: { channelConnection: true }
+		});
+		if (privConv) {
+			return privConv;
+		}
+		return await this.prisma.channel
+			.create({
+				data: {
+					name: convName,
+					kind: ChannelKind.DIRECT,
+					channelConnection: {
+						create: [
+							{ userId: lowUserId, role: ChannelRole.DEFAULT },
+							{ userId: highUserId, role: ChannelRole.DEFAULT }
+						]
+					}
+				}
+			})
+			.catch(() => {
+				throw new BadRequestException('Cannot create conversation', {
+					description: 'Your partner is not online'
+				});
+			});
+		//TODO: Define a new even for privmsg
+		void socketId;
+		return null;
```

Comment:
I'll try to re-write it!
I first couldn't because I did not implement the "unique name" so It was not possible. Now that I did do it I should have rewrote the entire function indeed.